Lab5/ScannerIntegration/Parser.py

```diff
@@ -0,0 +1,233 @@
+from Lab5.ScannerIntegration.Grammar import Grammar
+from Lab5.ScannerIntegration.Scanner import Scanner
+
+
+class ParserRecursiveDescent:
+    # input must be reversed for the pop to work
+    # such that the expansion of the work stack to be appended to the input list
+    def __init__(self, grammarTextLocation):
+        self.grammar = Grammar(grammarTextLocation)
+        self.work = []
+        self.input = []
+        self.state = "q"
+        self.index = 0
+        self.debug = True
+        self.tree = []
+        self.filename = grammarTextLocation
+        self.tokenCodes = {}
+        self.getTokens("tokens.in")
+        self.word = ""
+        self.iteration = 0
+
+    def getTokens(self, fileName):
+        with open(fileName, "r") as file:
+            for line in file:
+                tokens = line.strip().split(" ")
+                self.tokenCodes[tokens[0]] = int(tokens[2])
+
+    def printParserStep(self, step):
+        print("~~~~~~~~~~~~")

```

Comment:
Prints in the console are kinda confusing. 
Write when there is a Syntax error.
Print iterations better detailed.
Should delimit when lexic ends and syntax starts.